header/p2p/session.go

```diff
@@ -125,7 +148,17 @@ func (s *session) doRequest(
 
 	h, err := s.processResponse(r)
 	if err != nil {
-		s.errCh <- err
+		switch err {
+		case header.ErrNotFound:
+		default:
+			s.peerTracker.blockPeer(stat.peerID)
```

Comment:
Just in general. We need to be really careful with that and not block peers for the wrong reasons. E.g., our protocol allows `header.ErrMessageLimitExceeded` error, but we don't check for it in the switch.

Reply:
Our client checks the size of a message before sending it. So, It's not possible to get such an error in a normal case.

Reply:
Why do we need this error than to be a status code?

Reply:
Our server can handle requests from external clients.

Reply:
WDYM?

Reply:
I mean that anyone can write a custom client and open the stream with our server and start requesting headers. I think that is the reason why did we add this status code. On the other side, our client checks if the range is less than the limit and immediately returns an error to avoid any networking staff if we already know that this range request won't be handled.

Reply:
> I mean that anyone can write a custom client and open the stream with our server and start requesting headers

This means this custom client is misbehaving according to the protocol, and in other shrex protocols we recently went with the approach where in case of misbehavior, we just reset a stream without telling why. Mainly because misbehaviors are "speced out". Additionally, there is no need to be nice about it and send a status code, because there is no actual usage for this error code, really our client does nothing about the status code.  Discuss!

N.B. I understand why we need to have the Go error for the client side and I was my suggestion, the point is about the error code on the protocol level. It seems like an additional bloat to the protocol without real usage, which we may want cleanup, not in this PR ofc.

Reply:
Why can't we respond to this client about limitations? So it will be able to send a request with less range.

Reply:
Stream reset in this case is not the best solution IMO

Reply:
> Why can't we respond to this client about limitations?

But it does not do so, right? And how is it supposed to ___guess___ the right size if the server does not give it?